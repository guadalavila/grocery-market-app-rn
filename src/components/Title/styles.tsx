import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        marginVertical: 8,
        marginHorizontal: Sizes.M,
        paddingHorizontal: Sizes.XS,
    },
    label: {
        fontWeight: 'bold',
        fontSize: 20,
        color: Colors.black,
    },
    textRight: {
        color: Colors.primary,
        fontWeight: '600',
        textDecorationLine: 'underline',
        fontSize: 14,
    },
});
