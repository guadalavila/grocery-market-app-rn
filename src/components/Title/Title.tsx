import React from 'react';
import { Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { GlobalStyles } from '../../utils/GlobalStyles';
import { styles } from './styles';

interface ITitleProps {
    label: string;
    textRight: string;
    onPress: () => void;
}
const Title = ({ label, textRight, onPress }: ITitleProps) => {
    return (
        <View style={[GlobalStyles.row, GlobalStyles.justBtw, styles.container]}>
            <Text style={styles.label}>{label}</Text>
            <TouchableOpacity activeOpacity={0.7} onPress={onPress}>
                <Text style={styles.textRight}>{textRight}</Text>
            </TouchableOpacity>
        </View>
    );
};

export default Title;
