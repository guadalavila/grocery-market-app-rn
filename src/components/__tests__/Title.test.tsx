import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import Title from '../Title/Title';
describe('test Title component', () => {
    const onPressMock = jest.fn();
    const label = 'Hello World!';
    const textRight = 'View More';

    test('should match snapshot', () => {
        const title = render(<Title label={label} textRight={textRight} onPress={onPressMock} />);
        expect(title).toMatchSnapshot();
    });

    test('should render title', () => {
        render(<Title label={label} textRight={textRight} onPress={onPressMock} />);
        expect(screen.getByText(label)).toBeTruthy();
        expect(screen.getByText(textRight)).toBeTruthy();
    });

    test('should render button and call fn', () => {
        render(<Title label={label} textRight={textRight} onPress={onPressMock} />);
        const btn = screen.getByTestId('button');
        expect(btn).toBeTruthy();
        fireEvent.press(btn);
        expect(onPressMock).toHaveBeenCalled();
    });
});
