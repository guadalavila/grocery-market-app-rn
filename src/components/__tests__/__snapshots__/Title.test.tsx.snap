// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Title component should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "space-between",
      },
      Object {
        "marginHorizontal": 16,
        "marginVertical": 8,
        "paddingHorizontal": 4,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Hello World!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <Text
      style={
        Object {
          "color": "#68B984",
          "fontSize": 14,
          "fontWeight": "600",
          "textDecorationLine": "underline",
        }
      }
    >
      View More
    </Text>
  </View>
</View>
`;
